File: back/src/main/java/com/integradora/back/controller/orden/dto/OrdenPreviewDTO.java
package com.integradora.back.controller.orden.dto;

import lombok.AllArgsConstructor;
import lombok.Data;

import java.math.BigDecimal;

@Data
@AllArgsConstructor
public class OrdenPreviewDTO {
    private BigDecimal subtotal;
    private BigDecimal descuentoPromo;
    private BigDecimal descuentoPuntos;
    private BigDecimal montoDescuento;
    private String codigoPromoAplicado;
    private String tituloPromoAplicada;
    private BigDecimal total;
    private Integer puntosGanados;
}

File: mobile/app/src/main/java/proyecto/personal/proyectointegradorii/data/remote/dto/orden/OrdenPreviewDTO.kt
package proyecto.personal.proyectointegradorii.data.remote.dto.orden

data class OrdenPreviewDTO(
    val subtotal: Double,
    val descuentoPromo: Double,
    val descuentoPuntos: Double,
    val montoDescuento: Double,
    val codigoPromoAplicado: String?,
    val tituloPromoAplicada: String?,
    val total: Double,
    val puntosGanados: Int
)